test/chart/AreaChart.spec.tsx

```diff
@@ -237,6 +237,34 @@ describe('AreaChart', () => {
       expect(rightBrushTraveler?.firstChild).toHaveAttribute('x', '65');
       expect(container.querySelectorAll('.recharts-area-dot')).toHaveLength(1);
     });
+
+    it('should fire mouse events', () => {
+      const handleClickMock = vi.fn();
+      const handleDoubleClickMock = vi.fn();
+      const handleContextMenuMock = vi.fn();
+
+      const { container } = render(
+        <AreaChart
+          width={100}
+          height={50}
+          data={data}
+          onContextMenu={handleContextMenuMock}
+          onClick={handleClickMock}
+          onDoubleClick={handleDoubleClickMock}
+        >
+          <Area type="monotone" dataKey="uv" stroke="#ff7300" fill="#ff7300" />
+        </AreaChart>,
+      );
+
+      const surface = container.querySelector('.recharts-wrapper');
+      fireEvent.click(surface);
+      fireEvent.doubleClick(surface);
+      fireEvent.contextMenu(surface);
+
+      expect(handleClickMock).toHaveBeenCalledTimes(1);
+      expect(handleDoubleClickMock).toHaveBeenCalledTimes(1);
+      expect(handleContextMenuMock).toHaveBeenCalledTimes(1);
+    });
```

Comment:
one more thing! Can you also assert that there is something sent to the callback?
You don't have to assert every item

just something like

```
expect(handleDoubleClickMock).toHaveBeenCalledWith(expect.any(object));
expect(handleContextMenuMock).toHaveBeenCalledWith(expect.any(object));
```

Reply:
Oh yeah sure, will do.
Do you also want me to replicate this test in every other chart?

Reply:
I wouldn't require it, but if its not that much more effort you could totally do something like this

https://github.com/recharts/recharts/blob/50da69ec3e2cd8633159f50b18b0bed74f9fb1ff/test/component/Tooltip/Tooltip.sync.spec.tsx#L268-L288

make test cases which you can run in a loop, render the correct test case and the assertions should be the same for all of them

Reply:
That should be better now.